specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/StaticSites.json

```diff
@@ -0,0 +1,1916 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/StringDictionary"
+            }
+          },
+          "202": {
+            "description": "Asynchronous operation in progress.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/StringDictionary"
+            }
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Creates or updates the function app settings of a static site build.": {
+            "$ref": "./examples/CreateOrUpdateStaticSiteBuildFunctionAppSettings.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/staticSites/{name}/builds/{prId}/functions": {
+      "get": {
+        "tags": [
+          "StaticSites"
+        ],
+        "summary": "Gets the functions of a particular static site build.",
+        "description": "Description for Gets the functions of a particular static site build.",
+        "operationId": "StaticSites_ListStaticSiteBuildFunctions",
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "description": "Name of the static site.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "prId",
+            "in": "path",
+            "description": "The stage site identifier.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/StaticSiteFunctionOverviewCollection"
+            }
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Gets the functions of a particular static site build": {
+            "$ref": "./examples/ListStaticSiteBuildFunctions.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/staticSites/{name}/builds/{prId}/listFunctionAppSettings": {
```

Comment:
If functionAppSettings is its type, what will GET return and what will this POST return? there is weirdness in how these things are currently modeled.

Reply:
so similar to the users, these app settings can contain sensitive info so we don't want contributors to be able to do GET's by default. So we made them an action to list them so the owner of the app can do RBAC.